### PR TITLE
Better LoC scheme identification

### PR DESCRIPTION
**Pull request by paul-butcher**
https://github.com/wellcomecollection/catalogue-pipeline/issues/2298

### CODE REVIEW

File: pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraConceptIdentifier.scala [hunk 1]
```diff
@@ -22,17 +23,63 @@ import weco.sierra.models.marc.VarField
...
+      // LCSH in the Works Catalogue, but these appear to be errors, so an exact match is
+      // enforced here in order to guard against such typos re-emerging.
+      case "sh" => IdentifierType.LCSubjects
+      // There are actually seven different prefixes in use in the whole LCNames Authority file,
+      // but because there is no definitive list of LCNames prefixes, we do not wish to accidentally
+      // exclude a real one in the future by insisting that the prefix we find must be a member of
+      // a closed list.
+      case prefix if prefix.head == 'n' => IdentifierType.LCNames
+      // Any prefix other than sh or n is an error. Common mistakes include MeSH ids marked as LoC ids
+      // and general typographical errors such as `shsh`.  Guard against these by rejecting the identifier.
```
Comment (paul-butcher) [inline pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraConceptIdentifier.scala:65]: // At time of writing, there were 65 examples of identifiers designated as LoC ids
      // that do not conform to this sh|n prefix convention.
      // They were all incorrect, mostly they were MeSH ids with an incorrect indicator2 value, but also urls and typos like the one in the comment.
      // Therefore, the best treatment is to reject them and get them fixed at source.

File: pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraConceptIdentifier.scala [hunk 2]
```diff
@@ -22,17 +23,63 @@ import weco.sierra.models.marc.VarField
...
+    *
+    */
+  private def locScheme(idValue: String): IdentifierType =
+    idValue.split("\\d", 2).head match {
+      // sh is the only legal prefix for a Subject Headings identifier.
+      // At time of writing, there were some other s~ prefixed identifiers in use and marked as
+      // LCSH in the Works Catalogue, but these appear to be errors, so an exact match is
+      // enforced here in order to guard against such typos re-emerging.
+      case "sh" => IdentifierType.LCSubjects
+      // There are actually seven different prefixes in use in the whole LCNames Authority file,
```
Comment (paul-butcher) [inline pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraConceptIdentifier.scala:59]: At time of writing there were five different n.+ prefixes in use in Wellcome data.
All of them are correct LCNames prefixes.

File: pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraConceptIdentifier.scala [hunk 3]
```diff
@@ -22,17 +23,63 @@ import weco.sierra.models.marc.VarField
...
+    * 0 - Library of Congress Subject Headings
+    *  Subject added entry conforms to and is appropriate for use in the Library of Congress Subject Headings (LCSH) and the Name authority files that are maintained by the Library of Congress.
+    *
+    */
+  private def locScheme(idValue: String): IdentifierType =
+    idValue.split("\\d", 2).head match {
+      // sh is the only legal prefix for a Subject Headings identifier.
+      // At time of writing, there were some other s~ prefixed identifiers in use and marked as
+      // LCSH in the Works Catalogue, but these appear to be errors, so an exact match is
+      // enforced here in order to guard against such typos re-emerging.
```
Comment (paul-butcher) [inline pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraConceptIdentifier.scala:57]: Incorrect prefixes I encountered include `shsh` `subjectsh` and `subject` (though I think the latter two may be from TEI)

**Review by jamieparkinson — APPROVED** [review]
Very nice sleuthing - would it be useful to turn your additional (PR) comments into (code) comments?

**Comment by paul-butcher** [comment]
> Very nice sleuthing - would it be useful to turn your additional (PR) comments into (code) comments?

They were originally code comments, but I thought them too tightly tied to the current moment, so I've tried to convey the essence of them without too much of the specifics in the code comments.

**Pull request merged**